app/src/ui/repositories-list/group-repositories.ts

```diff
@@ -89,7 +89,7 @@ export function groupRepositories(
       const { aheadBehind, changedFilesCount } =
         localRepositoryStateLookup.get(r.id) || fallbackValue
       const repositoryText =
-        r instanceof Repository ? [r.name, nameOf(r)] : [r.name]
+        r instanceof Repository ? [r.alias ?? r.name, nameOf(r)] : [r.name]
```

Comment:
This basically makes sure the alias is used to look for matches when the user filters the repository list.